File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        chars = ""
        diff = 0
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                diff += 1
                if not chars:
                    chars = f"{s1[i]}{s2[i]}"
                elif chars != f"{s2[i]}{s1[i]}":
                    return False
        return diff == 0 or diff == 2

File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        int prev = -1, swaps = 0;
        for(int i = 0; i < s1.length(); i++){
            if(s1.charAt(i) != s2.charAt(i)){
                if(prev != -1 && (
                    s1.charAt(prev) != s2.charAt(i) || 
                    s2.charAt(prev) != s1.charAt(i)
                    )
                  )
                    return false;
                prev = i;
                swaps++;
            }
        }
        return swaps == 2 || swaps == 0 ? true : false;
    }
}